src/lib/io/sol-iio.c

```diff
@@ -1054,21 +1054,21 @@ sol_iio_read_channel_value(struct sol_iio_channel *channel, double *value)
     if (!r) {
         SOL_WRN("Could not read channel [%s] in device%d", channel->name,
             device->device_id);
-        return false;
+        return -EBADMSG;
```

Comment:
Why `EBADMSG`? If `craft_filename_path` failed is more likely `ENOMEM` or `ENOENT`... but maybe there's some new guideline...